### ISSUE TITLE
pom.xml incorrectly sets source language level to 17 when it should be 11

### ISSUE THREAD

**Issue by lukedegruchy** [hapifhir/hapi-fhir#4031]
**Describe the bug**
The pom.xml incorrectly sets the source language level to 17 when it should be 11.  The testSource should remain at 17.

**To Reproduce**
Add a JDK 12+ language feature, for example, instanceof/cast pattern matching.

**Expected behavior**
The maven or IDE should fail, complaining about an unsupported language feature.

**Screenshots**
N/A

**Environment (please complete the following information):**
N/A

**Additional context**
N/A

**Comment by jamesagnew** [comment]
Looks like the source is 11 to me: https://github.com/hapifhir/hapi-fhir/blob/master/pom.xml#L866-L868

**Comment by jamesagnew** [comment]
If you mean the version that's in the `ide` profile at the bottom of the root pom.xml, that doesn't affect our built sources, it's just a workaround for an IDE limitation (more info in the comments right above that)

**Comment by lukedegruchy** [comment]
@jamesagnew Yes, I mean the version in. the IDE profile.  Sorry for not being clear.  I'll update the issue.

As for the comment, the linked issue seems to be fixed but the fixed version is not specified:

https://youtrack.jetbrains.com/issue/IDEA-85478

I'll try testing this and will update with what I find.

**Comment by jamesagnew** [comment]
Oh! It looks in that ticket looks like 2022.1 has the fix. Maybe this isn't needed after all... that's awesome if so
